src/t8_schemes/t8_standalone/t8_standalone_implementation.hxx

```diff
@@ -0,0 +1,1568 @@
+   * \note Not every element that is created in t8code will be created by a call
+   * to this function. However, if an element is not created using \ref t8_element_new,
+   * then it is guaranteed that \ref t8_element_init is called on it.
+   * \note In debugging mode, an element that was created with \ref t8_element_new
+   * must pass \ref t8_element_is_valid.
+   * \note If an element was created by \ref t8_element_new then \ref t8_element_init
+   * may not be called for it. Thus, \ref t8_element_new should initialize an element
+   * in the same way as a call to \ref t8_element_init would.
+   * \see t8_element_init
+   * \see t8_element_is_valid
+   */
+  /* TODO: would it be better to directly allocate an array of elements,
+   *       not element pointers? */
+  void
+  element_new (int length, t8_element_t **elem) const
+  {
+    /* allocate memory */
+    T8_ASSERT (this->scheme_context != NULL);
+    T8_ASSERT (0 <= length);
+    T8_ASSERT (elem != NULL);
+
+    for (int i = 0; i < length; ++i) {
+      elem[i] = (t8_element_t *) sc_mempool_alloc ((sc_mempool_t *) this->scheme_context);
+    }
+
+/* in debug mode, set sensible default values. */
+#ifdef T8_ENABLE_DEBUG
+    {
+      int i;
+      for (i = 0; i < length; i++) {
+        t8_standalone_scheme<TEclass>::element_init (1, elem[i]);
+      }
+    }
+#endif
+  }
+
+  /** Initialize an array of allocated elements.
+   * \param [in] length   The number of elements to be initialized.
+   * \param [in,out] elems On input an array of \b length many allocated
+   *                       elements.
+   * \param [in] called_new True if the elements in \a elem were created by a call
+   *                       to \ref t8_element_new. False if no element in \a elem
+   *                       was created in this way. The case that only some elements
+   *                       were created by \ref t8_element_new should never occur.
+   * \note In debugging mode, an element that was passed to \ref t8_element_init
+   * must pass \ref t8_element_is_valid.
+   * \note If an element was created by \ref t8_element_new then \ref t8_element_init
+   * may not be called for it. Thus, \ref t8_element_new should initialize an element
+   * in the same way as a call to \ref t8_element_init would.
+   * Thus, if \a called_new is true this function should usually do nothing.
+   * \see t8_element_new
+   * \see t8_element_is_valid
+   */
+  static inline void
+  element_init (int length, t8_element_t *elem)
+  {
+#ifdef T8_ENABLE_DEBUG
+    int i;
+    t8_standalone_element<TEclass> *el = (t8_standalone_element<TEclass> *) elem;
+    /* Set all values to 0 */
+    for (i = 0; i < length; i++) {
+      t8_standalone_scheme<TEclass>::element_set_linear_id ((t8_element_t *) (el + i), 0, 0);
+      T8_ASSERT (t8_standalone_scheme<TEclass>::element_is_valid ((t8_element_t *) (el + i)));
+    }
+#endif
+  }
+
+  static inline void
+  element_deinit (int length, t8_element_t *elem)
```

Comment:
```suggestion
  element_deinit (const int length, t8_element_t *elem)
```

Reply:
why is this function not implemented?

Reply:
Because nothing is allocated in elements in this scheme, so nothing needs to be deallocated